src/helpers/measureTitleHelper.ts

```diff
@@ -56,113 +43,179 @@ function findOverTimeComparisonType(measureDefinitionType: IMeasureDefinitionTyp
     return OverTimeComparisonTypes.NOTHING;
 }
 
+function containsMeasureTitleItem(measureTitleProps: IMeasureTitleProps[], localIdentifier: string): boolean {
+    return measureTitleProps.some(prop => prop.localIdentifier === localIdentifier);
+}
+
+function containsMeasureTitleItems(measureTitleProps: IMeasureTitleProps[], localIdentifiers: string[]): boolean {
+    return localIdentifiers.every(identifier => containsMeasureTitleItem(measureTitleProps, identifier));
+}
+
+function findMeasureTitleItem(measureTitles: IMeasureTitleProps[], localIdentifier: string): IMeasureTitleProps | null {
+    return measureTitles.find(prop => prop.localIdentifier === localIdentifier) || null;
+}
+
 function findTitleForDerivedMeasure(
     measureDefinitionType: IMeasureDefinitionType,
-    measureBucketItems: IMeasure[],
+    measureTitleProps: IMeasureTitleProps[],
     suffixFactory: DerivedMeasureTitleSuffixFactory
-) {
+): string {
     const masterMeasureIdentifier = getMasterMeasureIdentifier(measureDefinitionType);
     if (masterMeasureIdentifier === null) {
         return undefined;
     }
-    const masterMeasure = getMasterMeasure(measureBucketItems, masterMeasureIdentifier);
-    const derivedMeasureTitleBase = getDerivedMeasureTitleBase(masterMeasure);
+
+    const measureProps = findMeasureTitleItem(measureTitleProps, masterMeasureIdentifier);
```

Comment:
This method can return null but the result is used without checking for the nullability.